### PR TITLE
fix: (Platform) add bug fixes for Upload Collection

### PR DESCRIPTION
**Pull request by Betrozov**
#### Please provide a link to the associated issue.
Fixes https://github.com/SAP/fundamental-ngx/issues/4032
Fixes https://github.com/SAP/fundamental-ngx/issues/4037
Fixes https://github.com/SAP/fundamental-ngx/issues/4038
Fixes https://github.com/SAP/fundamental-ngx/issues/4039
Fixes https://github.com/SAP/fundamental-ngx/issues/4043
Fixes https://github.com/SAP/fundamental-ngx/issues/4053
Fixes https://github.com/SAP/fundamental-ngx/issues/4033
Fixes https://github.com/SAP/fundamental-ngx/issues/4041

#### Please provide a brief summary of this pull request.
Addressed issues
#### Please check whether the PR fulfills the following requirements

- [X] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

### CODE REVIEW

**Review by KevinOkamoto — APPROVED** [review]
LGTM

File: apps/docs/src/app/platform/component-docs/platform-upload-collection/platform-upload-collection-examples/platform-upload-collection-base-data-provider.ts [hunk 1]
```diff
@@ -18,11 +18,16 @@ import {
 } from '@fundamental-ngx/platform';
 
 import { generateUploadCollectionItems } from './platform-upload-collection-items-generator';
+import { HttpClient } from '@angular/common/http';
 
 export class PlatformUploadCollectionDataProviderExample extends UploadCollectionDataProvider {
     items: UploadCollectionItem[] = generateUploadCollectionItems(50, 4, 2);
     private _cancelUploadNewFileIds: (string | number)[] = [];
```
Comment (InnaAtanasova) [inline apps/docs/src/app/platform/component-docs/platform-upload-collection/platform-upload-collection-examples/platform-upload-collection-base-data-provider.ts:25]: I know these are not part of the PR :D but I noticed they are missing the typedocs

File: libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts [hunk 2]
```diff
@@ -383,7 +383,7 @@ export class UploadCollectionComponent {
         const dialogRef = this._dialogService.open(NewFolderComponent, {
             responsivePadding: true,
             backdropClickCloseable: false,
-            height: '70%',
+            height: '320px',
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts:386]: maybe you want to use rem?

**Review by InnaAtanasova — APPROVED** [review]
LGTM

File: apps/docs/src/app/platform/component-docs/platform-upload-collection/platform-upload-collection-examples/platform-upload-collection-base-data-provider.ts (hunk 1 above)
Comment (InnaAtanasova) [reply apps/docs/src/app/platform/component-docs/platform-upload-collection/platform-upload-collection-examples/platform-upload-collection-base-data-provider.ts:25]: oh, it's the example! Sorry, you can ignore my comment. Should not review PRs before coffee.

**Review by salarenko — APPROVED** [review]

**Review by mykolavorobiov — APPROVED** [review]
